components/builder-depot/src/server.rs

```diff
@@ -1515,6 +1542,13 @@ pub fn routes<M: BeforeMiddleware + Clone>(insecure: bool, basic: M, worker: M)
                 XHandler::new(upload_package).before(basic.clone())
             }
         },
+        package_upload_bldr: post "/pkgs/internal/:origin/:pkg/:version/:release" => {
+            if insecure {
+                XHandler::new(upload_package_bldr)
+            } else {
+                XHandler::new(upload_package_bldr).before(basic.clone())
+            }
+        },
         schedule: post "/pkgs/schedule/:origin/:pkg" => {
```

Comment:
should we add a `schedule_build` query parameter to the url and then handle this all in the same handler?